compiler-core/src/type_.rs

```diff
@@ -422,6 +434,121 @@ impl Type {
             _ => None,
         }
     }
+
+    #[must_use]
+    /// Returns `true` is the two types are the same. This differs from the
+    /// standard `Eq` implementation as it also follows all links to check if
+    /// two types are really the same.
+    ///
+    pub fn same_as(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Type::Named { .. }, Type::Fn { .. } | Type::Tuple { .. }) => false,
+            (one @ Type::Named { .. }, Type::Var { type_ }) => {
+                type_.as_ref().borrow().same_as_other_type(one)
+            }
+            (
+                Type::Named {
+                    package,
+                    module,
+                    name,
+                    ..
+                },
+                Type::Named {
+                    package: other_package,
+                    module: other_module,
+                    name: other_name,
+                    ..
```

Comment:
There's a bug here! The parameters must also be the same

Reply:
Oops thanks for catching this!